proxy/Main.cc

```diff
@@ -460,7 +460,7 @@ proxy_signal_handler(int signo, siginfo_t *info, void *ctx)
   shutdown_event_system = true;
   sleep(1);
 
-  ::exit(signo);
```

Comment:
I couldn't understand why this change can cause an issue with leak detection.I read your original ticket about this line changing from `_exit` to `exit`. The reason you made that change was to fix some crashes on fedora instead of fixing leak detection issues. However,
it's never okay to call `exit` in a signal handler.  `exit()` is not an Async-signal-safe function, which can refer to [this man page](http://man7.org/linux/man-pages/man7/signal.7.html). Hence, we got lots of crashes when stopping or restarting ATS.
Besides, @zwoop tested this patch didn't make any behavior change of leak detection tools.